enhancements/sig-policy/54-core-policy-module/README.md

```diff
@@ -0,0 +1,227 @@
+
+#### Story 2
+
+As a creator of a new policy controller, I can focus on my controller's 
+specialized detection of violations, and rely on common implementations of
+functions in order to interact with the policy framework.
+
+#### Story 3
+
+As a developer of multiple policy controllers, I can fix a bug or add a feature
+to common policy behavior in one place, and not duplicate it in each controller.
+
+### Implementation Details/Notes/Constraints 
+
+Although kubebuilder will automatically pull API validation from embedded
+structs (like when types embed `metav1.TypeMeta`), it will not automatically
+pull RBAC permissions on methods into the project. So RBAC requirements on
+common functions (eg `get namespaces` for selecting namespaces) will need to
+be well-documented.
+
+This module can include features that aren't strictly required or currently used
+by the policy framework. For example, storing additional structured information
+about k8s objects that the policy looks at in a `status.relatedObjects` field.
+Experimental features can be added, but should be marked as such by the
+versioning of the repository.
+
+### Risks and Mitigation
+
+(unknown)
+
+## Design Details
+
+Here is a part of the common struct and validation from a
+[prototype](https://github.com/JustinKuli/policycore-test/blob/main/api/v1/policycore_types.go)
+of this proposal:
+
+```golang
+type PolicyCoreSpec struct {
+	Severity          Severity          `json:"severity,omitempty"`
+	RemediationAction RemediationAction `json:"remediationAction,omitempty"`
+	NamespaceSelector NamespaceSelector `json:"namespaceSelector,omitempty"`
+}
+
+//+kubebuilder:validation:Enum=low;medium;high;critical
+type Severity string
+
+const (
+	LowSeverity      Severity = "low"
+	MediumSeverity   Severity = "medium"
+	HighSeverity     Severity = "high"
+	CriticalSeverity Severity = "critical"
+)
+
+// ... other types omitted for brevity
+```
+
+With this struct, a new policy controller could implement their type like this,
+in order to get the same validation automatically:
+
+```golang
+import policycore "github.com/open-cluster-management-io/policycore/api/v1" // example import - this repo does not exist
+
+type MyNewPolicySpec struct {
+    policycore.PolicyCoreSpec `json:",inline"`
+
+    Foo string `json:"foo"`
+}
+```
+
+Common behavior could be implemented in the common library through methods on 
+these new types, for example, utilizing the `spec.namespaceSelector` field could
+be a method with this signature and doc:
+
+```golang
+// GetNamespaces fetches all namespaces in the cluster and returns a list of the
+// namespaces that match the NamespaceSelector. The client.Reader needs access
+// for viewing namespaces, like the access given by this kubebuilder tag:
+// `//+kubebuilder:rbac:groups=core,resources=namespaces,verbs=get;list;watch`
+func (sel NamespaceSelector) GetNamespaces(ctx context.Context, r client.Reader) ([]string, error) {
```

Comment:
No 😓 , the existing controllers use an object with `Include` and `Exclude` lists, each of which supports (some) wildcards, so you might have something like:
```yaml
spec:
  namespaceSelector:
    include: ["*"]
    exclude:
      - "kube-*"
      - "openshift*"
```

See doc in https://github.com/open-cluster-management-io/config-policy-controller and the implementation in https://github.com/open-cluster-management-io/config-policy-controller/blob/main/pkg/common/namespace_selection.go.

Improving the namespace selector to work with a `metav1.ListOptions{LabelSelector: ...}`, is one of my personal motivations for this proposal. But it would need to be backwards-compatible with what we already have.